aicsimageio/readers/ome_zarr_reader.py

```diff
@@ -0,0 +1,180 @@
+
+import xarray as xr
+from fsspec.implementations.local import LocalFileSystem
+from fsspec.spec import AbstractFileSystem
+
+from .. import constants, exceptions, types
+from ..dimensions import DimensionNames
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from .reader import Reader
+
+try:
+    from ome_zarr.io import parse_url
+    from ome_zarr.reader import Reader as ZarrReader
+
+except ImportError:
+    raise ImportError(
+        "ome_zarr is required for this reader. " "Install with `pip install 'ome_zarr'`"
+    )
+
+###############################################################################
+
+
+class OmeZarrReader(Reader):
+    """
+    Wraps the ome-zarr-py API to provide the same aicsimageio Reader API but for
+    OmeZarr images.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    fs_kwargs: Dict[str, Any]
+        Any specific keyword arguments to pass down to the fsspec created filesystem.
+        Default: {}
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            _ = fs  # TODO: How to use the FS to test for suppored image
+            ZarrReader(parse_url(path, mode="r"))
+            return True
+
+        except ValueError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        channel_names: Optional[Union[List[str], List[List[str]]]] = None,
+        fs_kwargs: Dict[str, Any] = {},
+    ):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(
+            image,
+            enforce_exists=True,
+            fs_kwargs=fs_kwargs,
+        )
+
+        if not isinstance(self._fs, LocalFileSystem):
+            raise ValueError(
+                f"Cannot read OmeZarr from non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+        self._channel_names = channel_names
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            OmeZarr = ZarrReader(parse_url(self._path, mode="r")).zarr
+            self._scenes = tuple(
+                metadata_utils.generate_ome_image_id(i)
+                for i in range(len(OmeZarr.root_attrs["multiscales"]))
+            )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+        return self._xarr_format(delayed=True)
+
+    def _read_immediate(self) -> xr.DataArray:
+        return self._xarr_format(delayed=False)
+
+    def _xarr_format(self, delayed: bool) -> xr.DataArray:
+
+        OmeZarr = ZarrReader(parse_url(self._path, mode="r")).zarr
```

Comment:
This is kind of a nit but is it conventional to case `OmeZarr` like this? It reads like a class reference to me later on (as though the methods are accessed statically) but I may not be filled in on all Python conventions!

Reply:
fixed with [9c2ccf6](https://github.com/AllenCellModeling/aicsimageio/pull/493/commits/9c2ccf6506f33867fc773355c52a38d6c01b6c0e)